OpenRA.Game/Traits/Target.cs

```diff
@@ -133,6 +133,24 @@ public WPos CenterPosition
 			}
 		}
 
+		public CPos Location
+		{
+			get
+			{
+				switch (Type)
+				{
+					case TargetType.Actor:
+						return actor.Location;
+					case TargetType.FrozenActor:
+						return frozen.Location;
+					default:
+					case TargetType.Invalid:
+					case TargetType.Terrain:
+						throw new InvalidOperationException("Attempting to query the location of a Target which is neither Actor nor FrozenActor");
```

Comment:
The target code was specifically set up so that you could safely query the target `CenterPosition` even if the original target was killed (`TargetType.Actor` &rarr; `TargetType.Invalid`).  I'd like to avoid introducing different behaviour for this.